src/functions/utils/rxjs-dynamic-duration.ts

```diff
@@ -44,12 +44,55 @@ export const rxjsDynamicDuration = <T>(
 	mapDuration: (lastDuration: number, event: T, index: number) => number,
 	initialDuration = 0,
 ) => {
-	const mappableDuration = createMappableValue(mapDuration, initialDuration);
+	let index = 0;
+	let previousDuration: number | null = null;
 
 	// Return duration observable
 	return (value: T): Observable<T> => {
-		const delayTime = mappableDuration.getValue();
-		mappableDuration.next(value);
-		return timer(delayTime).pipe(mapTo(value), first());
+		const duration = isNull(previousDuration) ? initialDuration : mapDuration(previousDuration, value, index);
+		previousDuration = duration;
+		index++;
+		return timer(duration).pipe(mapTo(value), first());
 	};
 };
+
+/**
```

Comment:
This guy is intended to be similar to `debounce` and `debounceTime` to avoid surprises